test/specs/lib/factories-test.tsx

```diff
@@ -524,21 +534,18 @@ describe('factories', () => {
 
     describe('from an element', () => {
       itReturnsAValidElement(<div />)
+      itAppliesDefaultProps(<div />)
       itDoesNotIncludePropsFromMappedProp(<div />)
+      itMergesClassNames('element', 'user', { value: <div className="user" /> })
       itAppliesProps('element', { foo: 'foo' }, { value: <div {...{ foo: 'foo' } as any} /> })
-
-      test('forwards original element "as is"', () => {
-        testCreateShorthand(
-          {
-            Component: 'p',
-            value: (
-              <span {...{ commonProp: 'originalElement', originalElementProp: true } as any} />
-            ),
-            defaultProps: { commonProp: 'default', defaultProp: true },
-            overrideProps: { commonProp: 'override', overrideProp: true },
-          },
-          { commonProp: 'originalElement', originalElementProp: true },
-        )
+      itOverridesDefaultProps(
+        'element',
+        { some: 'defaults', overridden: false },
+        { some: 'defaults', overridden: true },
+        { value: <div {...{ overridden: true } as any} /> },
+      )
+      itOverridesDefaultPropsWithFalseyProps('element', {
+        value: <div {...{ undef: undefined, nil: null, zero: 0, empty: '' } as any} />,
```

Comment:
Do we need to create these superfluous objects and spread them like this?  It would be much simpler to just pass the props, if possible:

```suggestion
        value: <div undef={undefined} nil={null} zero={0} empty='' />,
```

Reply:
Typescript is complaining because these props are not valid for div element, that's why we have this object spread. I can create another object and spread it, but it is much easier indeed to read it if it is inline on the element.